server/server.go

```diff
@@ -301,189 +144,203 @@ type Sealer interface {
 	Unseal(*protocol.Operation) ([]byte, error)
 }
 
-type request struct {
-	pkt *protocol.Packet
-	// time just after the request was deserialized from the connection
-	reqBegin time.Time
-	connName string
+type handler struct {
+	name        string
+	outstanding int
+	s           *Server
+	c           *sync.Cond
```

Comment:
Instead of using a condition variable, a channel with a capacity of `maxConnPendingRequests` could be used. This way the `outstanding` variable can also be removed since we can just use the channel to indicate the available capacity.